mobile-browser-based-version/src/helpers/communication_script/helpers.js

```diff
@@ -116,18 +117,38 @@ export function dataReceivedBreak(recvBuffer, key, maxTries) {
  * @param {Array} arr 
  * @param {int} len 
  */
-export function checkArrayLen(arr, len) {
+export function checkArrayLenSync(arr, len) {
+    let maxTries = 100
     return new Promise((resolve) => {
-        (function waitData() {
-            console.log(arr.length)
-            if (arr.length >= len) {
+        (function waitData(n) {
+            if (arr.length >= len || maxTries <= n) {
                 return resolve();
             }
-            setTimeout(waitData, 100);
-        })();
+            setTimeout(() => waitData(n+1), 100);
+        })(0);
+    });
+}
+
+/**
+ * Waits until an array reaches a given length. Used to make 
+ * sure that all weights from peers are received.
+ * @param {Array} arr 
+ * @param {int} len 
+ */
+ export function checkArrayLenCommon(recvBuffer, len) {
```

Comment:
This seems like code duplication to me. Is it possible to have only `checkArrayLenSync` and call it with `Object.values(recvBuffer.avgWeights).flat(1)` to achieve the functionality of `checkArrayLenCommon` or have a flag that indicates if you need to call `Object.values(recvBuffer.avgWeights).flat(1)` inside?

Reply:
Before we used to have only `checkArrayLenSync` and we called it with `Object.values(recvBuffer.avgWeights).flat(1)` but the problem is that this is not a pointer to the `recvBuffer.avgWeights` anymore but it's a copy so its value will never change and hence even if a new weight is received you will not detect and you will wait for 10 seconds.

Reply:
I can add the flag but I don't really like it with the flag, and I don't have any better idea for a relatively quick fix. Do you?

Reply:
okay, I understand. Why don't you like it with a flag though? In my opinion, it is better than having two functions with 95% duplicated code.

Reply:
Yes definitely, I already changed it before, but I was asking wether you would have a better idea, sorry for the misunderstanding.